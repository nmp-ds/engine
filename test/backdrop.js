import { expect, test } from 'vitest';

import { setup } from './_helpers.js';

setup();

test('backdrop blur', async ({ uno }) => {
  const classes = [
    'backdrop-blur-none',
    'backdrop-blur',
    'backdrop-blur-s',
    'backdrop-blur-m',
    'backdrop-blur-l',
    'backdrop-blur-xl',
    'backdrop-blur-xxl',
    'backdrop-blur-xxxl',
  ];
  const antiClasses = [
    'backdrop-blur-',
    'backdrop-blur-4',
    'backdrop-blur-inherit',
    'backdrop-blur-sm',
    'backdrop-blur-lg',
    'backdrop-blur-xl2',
    'backdrop-blur-3xl',
  ];
  const { css } = await uno.generate([...classes, ...antiClasses]);
  expect(css).toMatchSnapshot();
});
