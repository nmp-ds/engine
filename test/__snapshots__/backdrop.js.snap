// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`backdrop blur 1`] = `
"/* layer: default */
.backdrop-blur{-webkit-backdrop-filter:blur(4px);backdrop-filter:blur(4px);}
.backdrop-blur-l{-webkit-backdrop-filter:blur(12px);backdrop-filter:blur(12px);}
.backdrop-blur-m{-webkit-backdrop-filter:blur(8px);backdrop-filter:blur(8px);}
.backdrop-blur-none{-webkit-backdrop-filter:blur(0);backdrop-filter:blur(0);}
.backdrop-blur-s{-webkit-backdrop-filter:blur(2px);backdrop-filter:blur(2px);}
.backdrop-blur-xl{-webkit-backdrop-filter:blur(16px);backdrop-filter:blur(16px);}
.backdrop-blur-xxl{-webkit-backdrop-filter:blur(24px);backdrop-filter:blur(24px);}
.backdrop-blur-xxxl{-webkit-backdrop-filter:blur(40px);backdrop-filter:blur(40px);}"
`;
